### ISSUE TITLE
reproducible on machine 100%: "UnhandledPromiseRejectionWarning: Error: write EPIPE" after simply visiting specific site and exiting

### ISSUE THREAD

**Issue by Mattwmaster58** [microsoft/playwright-python#483]
This is running on `playwright=0.171.0` on a GCP VM:

```
Linux instance-2 5.8.0-1020-gcp #21-Ubuntu SMP Tue Jan 19 10:14:24 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
```

I have installed all the libraries required to get chrome up and running. My code to reproduce the issue:
```py
with sync_playwright() as p:
    p.chromium.launch().newPage().goto("https://perks.uasu.ca")
```

The following error is generated upon exiting the with statement, and the script hangs (invariably):
```py
(node:11136) UnhandledPromiseRejectionWarning: Error: write EPIPE
    at afterWriteDispatched (internal/stream_base_commons.js:154:25)
    at writeGeneric (internal/stream_base_commons.js:145:3)
    at Socket._writeGeneric (net.js:786:11)
    at Socket._write (net.js:798:8)
    at doWrite (_stream_writable.js:403:12)
    at writeOrBuffer (_stream_writable.js:387:5)
    at Socket.Writable.write (_stream_writable.js:318:11)
    at Transport.send (/snapshot/playwright-cli/node_modules/playwright/lib/protocol/transport.js:43:25)
    at DispatcherConnection.onmessage (/snapshot/playwright-cli/lib/driver.js:49:61)
    at DispatcherConnection.sendMessageToClient (/snapshot/playwright-cli/node_modules/playwright/lib/dispatchers/dispatcher.
js:129:14)
(node:11136) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside o
f an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the no
de process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.h
tml#cli_unhandled_rejections_mode). (rejection id: 1)
(node:11136) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections tha
t are not handled will terminate the Node.js process with a non-zero exit code.
```

enabling `DEBUG=pw:api` provides this extra information:
```
DEBUG:uasumonitor.cli:15 session_id=None config_file=None
  pw:api navigating to "https://perks.uasu.ca/shop", waiting until "load" +0ms
  pw:api   navigated to "https://perks.uasu.ca/accounts/login/?next=/shop/" +373ms
  pw:api   "domcontentloaded" event fired +295ms
  pw:api   "load" event fired +186ms
```
When I navigate to a 404 on the same domain, the browser closes properly (same situation with loading any other of the random sites I've tested). No browser process is left running whether or not exiting the context manager throws an error. I can provide SSH access to the machine @pavelfeldman if need be (email on my profile).

**Comment by Mattwmaster58** [comment]
Not sure why I didn't see this on my windows machine, but I'm consistently seeing it now as well.

**Comment by mxschmitt** [comment]
I can't reproduce it on my local machine, try to use the latest Playwright version (beta: `pip install playwright==1.8.0a1`) and the API is not designed to be fluent. That means storing the browser, context, or page makes sense to clean them up afterwards. See this example, which works on my machine:

```py
from playwright.sync_api import sync_playwright

with sync_playwright() as p:
    browser = p.chromium.launch()
    page = browser.new_page()
    page.goto("https://perks.uasu.ca")
    page.screenshot(path="foo.png")
```

**Comment by Mattwmaster58** [comment]
> the API is not designed to be fluent.

I just did it that way for typing brevity :)

> Try to use normal Ubuntu 20 which is the recommended one.

for reference, my VM was hosted on Google Cloud with the Ubuntu 20.10 Minimal image.

The EPIPE issue that I observed was resolved with 1.8.0a1 _I think_ but my testing was cut short by another bug....
